pmd-core/src/main/java/net/sourceforge/pmd/rules/RuleBuilder.java

```diff
@@ -184,12 +184,12 @@ private void loadLanguageMinMaxVersions(Rule rule) {
         checkLanguageVersionsAreOrdered(rule);
     }
 
-    private void throwUnknownLanguageVersionException(String minOrMax, String unknownVersion) {
+    private void throwUnknownLanguageVersionException(String minOrMax, String unknownVersion, Language lang) {
```

Comment:
this change is unrelated, but the `language` attribute is not required in rule definition (may be provided by the rule implementation directly), so this may produce NPEs for definitions such as:

```xml
<!-- AccessorClassGenerationRule is an AbstractJavaRule, which sets language to Java, maximumLanguageVersion is invalid, but the builder is never explicitly told the language -->
<rule name="AccessorClassGeneration"
          since="1.04"
          maximumLanguageVersion="16"
          message="Avoid instantiation through private constructors from outside of the constructor's class."
          class="net.sourceforge.pmd.lang.java.rule.bestpractices.AccessorClassGenerationRule"
          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#accessorclassgeneration">
```

this bug just happened to bit me during my refactors :D

Reply:
I think this needs a test case. I'll add one when merging